jest-setup.js

```diff
@@ -23,162 +133,28 @@ global.testRule = (rule, schema) => {
-				schema.reject.forEach((testCase) => {
-					const spec = testCase.only ? it.only : it;
-
-					describe(`${util.inspect(schema.config)}`, () => {
-						describe(`${testCase.code}`, () => {
-							spec(testCase.description || 'no description', () => {
-								const options = {
-									code: testCase.code,
-									config: stylelintConfig,
-									syntax: schema.syntax,
-								};
-
-								return stylelint(options).then((output) => {
-									const actualWarnings = output.results[0].warnings;
-
-									expect(output.results[0].parseErrors).toEqual([]);
-									expect(actualWarnings).toHaveLength(
-										testCase.warnings ? testCase.warnings.length : 1,
-									);
-
-									if (
-										schema.fix &&
-										!testCase.fixed &&
-										testCase.fixed !== '' &&
-										!testCase.unfixable
-									) {
-										throw new Error(
-											'If using { fix: true } in test schema, all reject cases must have { fixed: .. }',
-										);
-									}
-
-									(testCase.warnings || [testCase]).forEach((expected, i) => {
-										const warning = actualWarnings[i];
-
-										expect(expected).toHaveMessage();
-
-										if (expected.message !== undefined) {
-											expect(_.get(warning, 'text')).toBe(expected.message);
-										}
-
-										if (expected.line !== undefined) {
-											expect(_.get(warning, 'line')).toBe(expected.line);
-										}
-
-										if (expected.column !== undefined) {
-											expect(_.get(warning, 'column')).toBe(expected.column);
-										}
-									});
-
-									if (!schema.fix) return;
-
-									// Check the fix
-									return stylelint(Object.assign({ fix: true }, options))
-										.then((output) => {
-											const fixedCode = getOutputCss(output);
-
-											if (!testCase.unfixable) {
-												expect(fixedCode).toBe(testCase.fixed);
-												expect(fixedCode).not.toBe(testCase.code);
-											} else {
-												// can't fix
-												if (testCase.fixed) {
-													expect(fixedCode).toBe(testCase.fixed);
-												}
-
-												expect(fixedCode).toBe(testCase.code);
-											}
-
-											return {
-												fixedCode,
-												warnings: output.results[0].warnings,
-											};
-										})
-										.then(({ fixedCode, warnings }) => {
-											// Checks whether only errors other than those fixed are reported.
-											return stylelint({
-												code: fixedCode,
-												config: stylelintConfig,
-												syntax: schema.syntax,
-											}).then((output) => ({
-												output,
-												warnings,
-											}));
-										})
-										.then(({ output, warnings }) => {
-											expect(output.results[0].warnings).toEqual(warnings);
-											expect(output.results[0].parseErrors).toEqual([]);
-										});
-								});
-							});
-						});
+				describe(`${util.inspect(schema.config)}`, () => {
+					describe(`${util.inspect(testCase.code)}`, () => {
```

Comment:
I think we need to preserve `\n`, `\r`, `\t` here.

Reply:
It will show `a\nb` instead of:

```
a
b
```